.azure/pipelines/jobs/default-build.yml

```diff
@@ -178,14 +178,11 @@ jobs:
         displayName: Install Node 14.x
         inputs:
           versionSpec: 14.x
-    - ${{ if eq(parameters.agentOs, 'Windows') }}:
-      - task: NuGetToolInstaller@1
-      - task: NuGetCommand@2
-        displayName: 'Clear NuGet caches'
-        condition: succeeded()
-        inputs:
-          command: custom
-          arguments: 'locals all -clear'
+    - task: NuGetCommand@2
```

Comment:
Are you sure this works?  Looking at the virtual environments repo, there's no nuget CLI installed on hosted linux agents and AFAIK not on ours either.

It may make more sense to just do `dotnet nuget locals all -clear` once a .NET Core SDK is installed.

Reply:
I see lots of failures in this PR but the errors are `Unable to locate executable file: 'mono'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.` _after_ the tool is found. That looks like a problem w/ the configuration of _some_ agents or in _some_ Docker images. Many of the Linux and macOS jobs look fine.

<details>
  <summary>Complete log</summary>

  ``` text
  Found tool in cache: NuGet 4.1.0 x64
  Found tool in cache: NuGet 4.1.0 x64
  Resolved from tool cache: 4.1.0
  Using version: 4.1.0
  Found tool in cache: NuGet 4.1.0 x64
  SYSTEMVSSCONNECTION exists true
  SYSTEMVSSCONNECTION exists true
  Detected NuGet version 4.1.0.2450 / 4.1.0
  SYSTEMVSSCONNECTION exists true
  ##[error]Error: Unable to locate executable file: 'mono'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
  ```

</summary>

Reply:
> It may make more sense to just do `dotnet nuget locals all -clear` once a .NET Core SDK is installed.

Is `dotnet` included in all container and agent configurations❔ If not, the SDK won't be installed until the build step that does the rest of our build.

Reply:
I'd forgotten about the `$env:NUGET_PACKAGES` overrides in the tools.* scripts. We shouldn't need any of this.